src/tdastro/astro_utils/passbands.py

```diff
@@ -2,257 +2,246 @@
-        """Calculate the in-band fluxes for all bands.
-
-        Note
-        ----
-        The given model needs to be interpolable or otherwise able to be evaluated with arbitrary wavelengths.
-
-        Parameters
-        ----------
-        model : PhysicalModel
-            A model object that can evaluate fluxes at arbitrary wavelengths.
-        times : np.ndarray
-            Array of times at which to evaluate the model.
+        flux_density_matrix : np.ndarray
+            A 2D array of flux densities where rows are times and columns are wavelengths.
+        wavelengths_angstrom : np.ndarray
+            An array of wavelengths in Angstroms corresponding to the columns of flux_density_matrix.
 
         Returns
         -------
         np.ndarray
-            A 2D array with in-band fluxes for all bands at all times.
+            An array of bandfluxes with length flux_density_matrix, where each element is the bandflux
+            at the corresponding time.
         """
-        # Find the union of all wavelengths in the bands
-        all_wavelengths = np.unique(
-            np.concatenate([self.normalized_system_response_tables[band][:, 0] for band in self.bands])
-        )
-
-        # Evaluate the model at all times and wavelengths
-        all_fluxes = model.evaluate(times, all_wavelengths)
-
-        # Prepare an array to hold the flux values
-        flux_matrix = np.zeros((len(times), len(self.bands)))
-
-        # Apply _get_in_band_flux to all fluxes
-        for i, band in enumerate(self.bands):
-            wavelengths_in_band = self.normalized_system_response_tables[band][:, 0]
+        # TODO: This area will be changed with incoming interpolation/exterpolation PR :)
+        # Check that the grid of wavelengths_angstrom matches the band's transmission table
+        # TODO
+        # For now, min exterpolation check: the bounds of the flux density grid are within the transmission
+        # table bounds
+        if (
+            wavelengths_angstrom[-1] < self.normalized_transmission_table[-1, 0]
+            or wavelengths_angstrom[0] > self.normalized_transmission_table[0, 0]
+        ):
+            raise ValueError(
+                f"Extrapolation needed: flux density interval [{wavelengths_angstrom[0]}, "
+                f"{wavelengths_angstrom[-1]}] is outside the transmission table interval "
+                f"[{self.normalized_transmission_table[0, 0]}, {self.normalized_transmission_table[-1, 0]}]."
+            )
 
-            # Find the indices of the wavelengths in the band in the union array
-            band_indices = np.searchsorted(all_wavelengths, wavelengths_in_band)
+        # Get only the flux densities that are within the passband's wavelength range
+        passband_wavelengths = self.normalized_transmission_table[:, 0]
+        passband_wavelengths_indices = np.searchsorted(wavelengths_angstrom, passband_wavelengths)
```

Comment:
I'm curious if it'll be better/faster to search for the min and max of the passband_wavelengths instead of all overlapping indices (since the wavelengths are supposed to be sorted).

Reply:
On a related note, should we check if the wavelengths are sorted when we load them? (It's likely good to check, but maybe that's the user's responsibility)

Reply:
- Good point--I'll use min and max for extrapolation check in the following PR. 
- I agree with you on sorted wavelengths--I recently put that in the function where I check the grid steps/if we need to interpolate at all; it's nice not to make users stay on top of that if we can do it too